packages/fetchai/protocols/fipa/dialogues.py

```diff
@@ -18,51 +18,36 @@
 # ------------------------------------------------------------------------------
 
 """
-This module contains the classes required for FIPA dialogue management.
+This module contains the classes required for fipa dialogue management.
 
 - DialogueLabel: The dialogue label class acts as an identifier for dialogues.
 - Dialogue: The dialogue class maintains state of a dialogue and manages it.
 - Dialogues: The dialogues class keeps track of all dialogues.
 """
 
 from enum import Enum
-from typing import Dict, List, Tuple, Union, cast
+from typing import Dict, Tuple, cast
 
 from aea.helpers.dialogue.base import Dialogue, DialogueLabel, Dialogues
 from aea.mail.base import Address
 from aea.protocols.base import Message
 
 from packages.fetchai.protocols.fipa.message import FipaMessage
 
-VALID_PREVIOUS_PERFORMATIVES = {
-    FipaMessage.Performative.CFP: [None],
-    FipaMessage.Performative.PROPOSE: [FipaMessage.Performative.CFP],
-    FipaMessage.Performative.ACCEPT: [FipaMessage.Performative.PROPOSE],
-    FipaMessage.Performative.ACCEPT_W_INFORM: [FipaMessage.Performative.PROPOSE],
-    FipaMessage.Performative.MATCH_ACCEPT: [
-        FipaMessage.Performative.ACCEPT,
-        FipaMessage.Performative.ACCEPT_W_INFORM,
-    ],
-    FipaMessage.Performative.MATCH_ACCEPT_W_INFORM: [
-        FipaMessage.Performative.ACCEPT,
-        FipaMessage.Performative.ACCEPT_W_INFORM,
-    ],
-    FipaMessage.Performative.INFORM: [
-        FipaMessage.Performative.MATCH_ACCEPT,
-        FipaMessage.Performative.MATCH_ACCEPT_W_INFORM,
-        FipaMessage.Performative.INFORM,
-    ],
-    FipaMessage.Performative.DECLINE: [
-        FipaMessage.Performative.CFP,
-        FipaMessage.Performative.PROPOSE,
-        FipaMessage.Performative.ACCEPT,
-        FipaMessage.Performative.ACCEPT_W_INFORM,
-    ],
-}  # type: Dict[FipaMessage.Performative, List[Union[None, FipaMessage.Performative]]]
+REPLY = {
+    "cfp": ["propose", "decline"],
+    "propose": ["accept", "accept_w_inform", "decline"],
+    "accept": ["decline", "match_accept", "match_accept_w_inform"],
+    "accept_w_inform": ["decline", "match_accept", "match_accept_w_inform", "inform"],
+    "decline": [],
+    "match_accept": [],
+    "match_accept_w_inform": ["inform"],
+    "inform": ["inform"],
+}
```

Comment:
Should be changed to Performative enum equivalent

Reply:
Also note, you inverted the meaning...so the implementation needs to take account of that